source/extensions/tracers/xray/tracer.cc

```diff
@@ -203,6 +209,10 @@ void Span::setTag(absl::string_view name, absl::string_view value) {
     // In this case, PeerAddress refers to the client's actual IP address, not
     // the address specified in the HTTP X-Forwarded-For header.
     http_request_annotations_.emplace(SpanXForwardedFor, ValueUtil::boolValue(false));
+  } else if (name == Error) {
+    if (value == True) {
```

Comment:
`if (name == Error && value == True)`

Reply:
True, we don't have a case where Error tag is sent with false value.

[situation 1](https://github.com/envoyproxy/envoy/blob/main/source/common/tracing/http_tracer_impl.cc#L108)
[situation 2](https://github.com/envoyproxy/envoy/blob/main/source/common/tracing/http_tracer_impl.cc#L255)

I am inclined to getting rid of the second if condition ~`if (value == True) `~ altogether would that be ok?

Reply:
Updated this change.
Now doing both name == Error & value == true check in the same line.